src/primitives/felt.ts

```diff
@@ -19,14 +19,14 @@ export class Felt {
   }
 
   add(other: MaybeRelocatable): Felt {
-    if (!(other instanceof Felt)) {
-      throw new PrimitiveError(ForbiddenOperation);
+    if (MaybeRelocatable.isFelt(other)) {
+      return new Felt((this.inner + other.inner) % Felt.PRIME);
     }
-    return new Felt((this.inner + other.inner) % Felt.PRIME);
+    throw new PrimitiveError(ForbiddenOperation);
   }
 
   sub(other: MaybeRelocatable): Felt {
-    if (!(other instanceof Felt)) {
+    if (!MaybeRelocatable.isFelt(other)) {
```

Comment:
The functions `isFelt()` and `isRelocatable()` are used for the class Felt in `felt.ts` but not for the class Relocatable in `relocatable.ts`, both should make use of it on methods with `MaybeRelocatable` parameters

Reply:
These assertions should all be in the same places in my opinion. Before, we had the `isFelt` inside the `Felt` class but it doesn't make much sense to put it there. It is preferable to put it in a dedicated `MaybeRelocatable` module because it is in this case of `MaybeRelocatable` that we are interested in knowing whether it is a `Felt` or a ` Relocatable`.